viewing_party/party.py

```diff
@@ -1,23 +1,142 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+    return None
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    updated_data = user_data.copy()
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            updated_data["watched"].append(movie)
+            updated_data["watchlist"].remove(movie)
+
+    return updated_data

```

Comment:
Great work in making a copy of `user_data`! Your approach of Iterating over the original data structure while modifying its copy avoids the indexing errors that usually occur when modifying a list while iterating over it (removing from user_data["watchlist"] as you're iterating through it).